sdk/cosmos/src/clients/cosmos_client.rs

```diff
@@ -51,52 +46,76 @@ impl CosmosOptions {
     }
 }
 
+/// Create a Pipeline from CosmosOptions
+fn new_pipeline_from_options(options: CosmosOptions) -> Pipeline {
```

Comment:
Nit: I personally like keeping the struct definition and the impl together which would mean moving these functions below the `impl` block. That's definitely not a blocker though.

Reply:
I'm fine with either. I'll change it to be in the impl.